### PR TITLE
Implemented payments, loyalty, discounts

### PR DESCRIPTION
**Pull request by dasAtRagedy**

### CODE REVIEW

File: Endpoints.cs [hunk 1]
```diff
@@ -646,4 +766,50 @@ private static void MapCompanyEndpoints(RouteGroupBuilder group)
...
+            })
+            .Accepts<PaymentInformation>("application/json")
+            .RequireAuth()
+            .Produces(StatusCodes.Status403Forbidden)
+            .Produces<Payment>(StatusCodes.Status201Created);
+
+        paymentsGroup.MapGet("{paymentId}/receipts", (string companyId, int paymentId) => Results.Ok())
+            .WithOpenApi(operation => new(operation)
+            {
+                Summary = "Generate a receipt for a payment",
```
Comment (BenasB) [inline Endpoints.cs:788]: A bit sus that `Get` "generates"

File: Endpoints.cs [hunk 2]
```diff
@@ -67,6 +68,17 @@ private static void MapOrdersEndpoints(RouteGroupBuilder group)
             .RequireAuth()
             .Produces(StatusCodes.Status403Forbidden);
 
+        ordersGroup.MapPost("{orderId}/discounts", (string companyId, int orderId) => Results.Ok())
+            .WithOpenApi(operation => new(operation)
+            {
+                Summary = "Create an order discount",
+            })
+            .Accepts<ItemInformation>("application/json")
```
Comment (BenasB) [inline Endpoints.cs:76]: Not sure about this accept, why would it need an `ItemInformation` as a request body

**Review by BenasB — APPROVED** [review]

File: Endpoints.cs (hunk 1 above)
Comment (6ar8nas) [reply Endpoints.cs:788]: Yeah, I think we should change this to `Fetch a receipt for a payment.` or similar.

Comment (dasAtRagedy) [reply Endpoints.cs]: fixed

File: Endpoints.cs (hunk 2 above)
Comment (dasAtRagedy) [reply Endpoints.cs]: fixed

**Pull request merged**